relay-server/src/services/processor.rs

```diff
@@ -1368,6 +1375,18 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
+    /// Whether the event in the envelope should be normalized.
+    ///
+    /// Under normal operation, envelopes should go through PoP relays and then
+    /// processing relays. However, during incidents, envelopes may go straight
+    /// to processing without going through PoPs. Events should be normalized in
+    /// both cases, but should not be normalized twice (for efficiency).
```

Comment:
Again, we should describe an abstract architecture we want to support independent of what we already have. A chain of trusted Relays in which certain things only need to happen once (at the first layer). Which is independent of whether something is a processing Relay or not.

Reply:
Replied to the comment above in https://github.com/getsentry/relay/pull/3377#discussion_r1555914322, resolving this one to keep the convo focused on the previous comment.